### PR TITLE
Toggle Shown/Hidden Table Column Settings

### PR DESCRIPTION
**Pull request by joneath**
Builds off of PR #4436

### CODE REVIEW

**Comment by mazameli** [comment]
This is a little whimsical, but how about this for the state where every field is hidden? I went back and forth on whether or not to allow the user to click `Done` if every field is hidden, but I think that as long as we're explicit that all the fields are hidden, then this feels less restrictive or fiddly.

hidden-fields

query with hidden fields


Here's the illustration asset in 1x and 2x:
[illustration.zip](https://github.com/metabase/metabase/files/1454952/illustration.zip)

**Comment by joneath** [comment]
This is ready to go except for the updated design referenced in #4436 which would require moving the business logic and markup from `ChartSettingOrderedFields.jsx` to `ChartSettings.jsx` which @mazameli seemed opposed to.

**Comment by mazameli** [comment]
Yeah, the reason I said this…
>So I'm not sure how to get the stacked checkbox on that same line short of putting its markup into ChartSettings.jsx, which I don't think we want to do.

…is just that it seemed to me (naively) that it might be putting a very specific element into what's supposed to be a generalized component. I think I'm okay with the UI as-is so that we can merge it and so people can use the feature, but I would like to solicit some thoughts from the front-end guys (@attekei, @tlrobinson, @kdoh) to see if they have any thoughts about improving `ChartSettings.jsx` so that a change like this would be possible.

**Comment by mazameli** [comment]
In case I was vague, I think this is okay to merge now. :)

File: frontend/src/metabase/visualizations/visualizations/Table.jsx [hunk 1]
```diff
@@ -125,21 +126,38 @@ export default class Table extends Component {
...
-            <TableComponent
-                {...this.props}
-                data={data}
-                isPivoted={isPivoted}
-                sort={sort}
-            />
-        );
+        if (isColumnsDisabled) {
+            return (
+                <div className={"flex-full px1 pb1 text-centered flex flex-column layout-centered " + (isDashboard ? "text-slate-light" : "text-slate")}>
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/visualizations/Table.jsx]: There's a utility called 'classnames' that you can use here to make this kind of conditional class application a bit easier.  If you look at https://github.com/metabase/metabase/pull/6339/files#diff-5114dfd896a1b8c2a72dd4e397024b01R68 you'll see an example.

**Review by kdoh — APPROVED** [review]
Left a minor comment on use of `classnames.` Once that gets updated this looks good to go from my perspective. Thanks for the tests!

**Pull request merged**